src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,182 @@
+"""This module provides integration with AWS AssumeRole functionality."""
+
+import datetime
+import hashlib
+import typing
+
+from awx_plugins.interfaces._temporary_private_django_api import (  # noqa: WPS436
+    gettext_noop as _,
+)
+
+import boto3
+from botocore.exceptions import ClientError
+
+
+if typing.TYPE_CHECKING:
+    from mypy_boto3_sts.client import STSClient
+    from mypy_boto3_sts.type_defs import (
+        AssumeRoleResponseTypeDef,
+        CredentialsTypeDef,
+    )
+
+from .plugin import CredentialPlugin
```

Comment:
:loudspeaker: Don't worry about `pylint` not liking this line. I just disabled that check on `devel` via 0b3155efcee6ebdef456b289d57ac661e9f4a824. It is unable to handle complicated `isort` configurations.